File: android/live-update-provider/src/main/kotlin/io/ionic/liveupdateprovider/ProviderError.kt
package io.ionic.liveupdateprovider

/** An error thrown by a live update provider. */
sealed class ProviderError(message: String, cause: Throwable? = null) : Exception(message, cause) {
    /** The provider received an invalid configuration. */
    class InvalidConfiguration(
        val details: String,
        cause: Throwable? = null
    ) : ProviderError("Invalid configuration: $details", cause)
}

File: ios/Sources/LiveUpdateProvider/ProviderError.swift
import Foundation


/// An error thrown by a live update provider.
public enum ProviderError: Error {
    /// The provider received an invalid configuration.
    case invalidConfiguration(message: String, underlyingError: Error?)
}

public extension ProviderError {
    static func invalidConfiguration(message: String) -> ProviderError {
        .invalidConfiguration(message: message, underlyingError: nil)
    }
}

extension ProviderError: LocalizedError {
    public var errorDescription: String? {
        switch self {
        case .invalidConfiguration(let message, _):
            return "Invalid configuration: \(message)"
        }
    }
}
